geniza/corpus/views.py

```diff
@@ -459,6 +463,67 @@ def get(self, request, *args, **kwargs):
         return JsonResponse(dict(annotation_list), encoder=iiif_utils.AttrDictEncoder)
 
 
+class DocumentMerge(FormView, PermissionRequiredMixin):
+    permission_required = ("corpus.change_document", "corpus.delete_document")
```

Comment:
@rlskoeser This view is open to the public, despite permissions being set. I tried updating to `permission_required = ("change_document", "delete_document")` after seeing the codenames related to `Permissions.objects.filter(codename__contains="document")` but that too didn't work. What's worse, incorrect permissions don't raise errors.  Any ideas?

Reply:
It's the order!  (One of the checks must get skipped if the permission mixin is not first...)

You need to switch them:
`class DocumentMerge(PermissionRequiredMixin, FormView):`

Reply:
Really?!? Yeah that was it 😄thanks for the help